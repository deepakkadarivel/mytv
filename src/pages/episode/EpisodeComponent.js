import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './_style.scss';

class EpisodeComponent extends Component {
    componentDidMount() {
        const { fetchEpisode } = this.props;
        fetchEpisode();
    }

    render() {
        const { episode } = this.props;
        const { url, imageAlt, title, season, number, airdate, desc } = episode;
        return (
            <div className="Episode container">
                <img className="Episode-image" src={url} alt={imageAlt} />
                <span className="Episode-content">
                    <span className="Episode-content--title">{title}</span>
                    <span className="Episode-content--episode">
                        Season: {season} Number: {number}
                    </span>
                    <span className="Episode-content--airDate">
                        Airdate: {airdate}
                    </span>
                    <span className="Episode-content--desc">{desc}</span>
                </span>
            </div>
        );
    }
}

EpisodeComponent.propTypes = {
    fetchEpisode: PropTypes.func.isRequired,
    episode: PropTypes.object.isRequired
};

export default EpisodeComponent;
