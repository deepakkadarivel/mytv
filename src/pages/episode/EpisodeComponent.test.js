import React from 'react';
import { shallow } from 'enzyme';
import EpisodeComponent from './EpisodeComponent';

const props = {
    episode: {
        url: 'http://placeholder.com/32',
        imageAlt: 'test image',
        title: 'show title',
        season: 1,
        number: 8,
        airdate: '1990',
        desc: ''
    },
    fetchEpisode: jest.fn()
};

test('renders fine', () => {
    const wrapper = shallow(<EpisodeComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
});
