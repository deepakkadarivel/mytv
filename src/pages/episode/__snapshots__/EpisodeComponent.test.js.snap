// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders fine 1`] = `ShallowWrapper {}`;
